### DISCUSSION TITLE
Infura and Truffle Inquiry

### DISCUSSION THREAD

**Discussion by jvel07** [hashlips-lab/nft-erc721-collection#436] [General]
@liarco, as usual, lovely work you do every day! Keep it up!

I am a bit confused with Infura and Truffle. Following your tutorials, Truffle is an option to not provide private keys, so you have to approve each transaction manually (deployment, WL-open, etc.). So is it a replacement for Infura?

- For launching a collection, do one strictly needs a node hosted in Infura?
- or is Truffle enough for that? --or is it just for testing? (although it's possible to deploy on mainnet by just switching the network on MM, so I'd assume it's not only for testing?)

**Comment by liarco** [comment]
Hi @jvel07, thx for your feedback!

Truffle is totally fine on mainnet, what it does is providing a fake Web3 provider that interacts with your web wallet instead of using your private key directly.

So:
* Infura (or any other private RPC service) is not required
* Truffle can be used on the mainnet by simply switching the network on your web wallet
